tests/test_deploy_github_pages.py

```diff
@@ -384,3 +391,111 @@ def test_infer_source_branch(setup_test_env):
+                     f"please commit and push the following files:\n .*")
+    assert e.type == SystemExit
+
+
+def test_abort_local_committed_changes_exist_in_source_branch(setup_test_env):
+    source_branch = "5-add-tests"
+    run(["git", "checkout", source_branch], check=True)
+    cwd = os.getcwd()
+    current_commit_id = run(["git", "rev-parse", "HEAD"], capture_output=True, text=True, check=True)
+    with open("./index.rst", "a") as file:
+        file.write("\n\nThis text is a change.")
+    run(["git", "add", "*"], check=True)
+    run(["git", "commit", "-m", "test-commit"], check=True)
+    new_current_commit_id = run(["git", "rev-parse", "HEAD"], capture_output=True, text=True, check=True)
+    print(current_commit_id)
+    print(new_current_commit_id)
+    target_branch = "test-docu-new-branch"
+
+    with pytest.raises(SystemExit) as e:
+        deploy_github_pages.deploy_github_pages(["--target_branch", target_branch,
+                                                 "--push_origin", "origin",
+                                                 "--push_enabled", "push",
+                                                 "--source_branch", source_branch,
+                                                 "--source_dir", cwd,
+                                                 "--module_path", ["../test_package", "../another_test_package"]])
+    assert e.match(f"Abort. Local commit id .* and commit id of source "
+                   f"branch remote .* are not equal. Please push your changes.")
+    assert e.type == SystemExit
+
+def test_abort_source_branch_only_exists_locally(setup_test_env):
+    a_branch = "5-add-tests"
+    run(["git", "checkout", a_branch], check=True)
+    temp_test_branch = "temp-test-branch"
+    run(["git", "checkout", "-B", temp_test_branch], check=True)
+    run(["git", "checkout", a_branch], check=True)
+    cwd = os.getcwd()
+    target_branch = "test-docu-new-branch"
+    with pytest.raises(SystemExit) as e:
+        deploy_github_pages.deploy_github_pages(["--target_branch", target_branch,
+                                                 "--push_origin", "origin",
+                                                 "--push_enabled", "push",
+                                                 "--source_branch", temp_test_branch,
+                                                 "--source_dir", cwd,
+                                                 "--module_path", ["../test_package", "../another_test_package"]])
+
+    assert e.match(f"Source branch exists locally, but not on remote, and source branch is not current branch."
+                   f"Please push your source branch to remote.")
+    assert e.type == SystemExit
+
+
+def test_abort_if_no_source_branch_detected(setup_test_env):
+    source_branch = "5-add-tests"
+    run(["git", "checkout", source_branch], check=True)
+    cwd = os.getcwd()
+    # break the local git repository
+    os.remove("../.git/HEAD")
```

Comment:
Is this test repeatable?

Reply:
It happens in a temp folder which is recreated for each time the test is run, so yes(If i understood your question correctly).